var/spack/repos/builtin/packages/c/package.py

```diff
@@ -14,17 +14,32 @@ class C(Package):
     homepage = "http://open-std.org/JTC1/SC22/WG14/www/standards"
     virtual = True
 
-    def test(self):
-        test_source = self.test_suite.current_test_data_dir
+    def test_c(self):
+        """Compile and run 'Hello world'"""
 
-        for test in os.listdir(test_source):
-            filepath = test_source.join(test)
-            exe_name = "%s.exe" % test
+        test_source = self.test_suite.current_test_data_dir
 
-            cc_exe = os.environ["CC"]
-            cc_opts = ["-o", exe_name, filepath]
-            compiled = self.run_test(cc_exe, options=cc_opts, installed=True)
+        cc_exe = os.environ["CC"]
+        cc_exe = which(join_path(self.prefix.bin, cc_exe))
+        if cc_exe is None:
+            raise SkipTest(f"{os.environ['CC']} not found in {self.version}")
```

Comment:
I don't *think* this can happen in that the environment variable is set to a valid path.